### PR TITLE
Add last urgent build to JSON response

### PR DESCRIPTION
**Pull request by doomspork**
This updates the build JSON response to include the last urgent build's ID in the `urgent` field.

### CODE REVIEW

**Comment by doomspork** [comment]
Happy to drop 18237a0 if we want. It adds the formatter config and runs it against the project.

File: lib/deckard/build.ex [hunk 1]
```diff
@@ -3,15 +3,46 @@ defmodule Deckard.Build do
...
+  defp get(key, channel) do
+    case Redis.query(["HGETALL", key]) do
       [] ->
         {:error, :not_found}
 
       raw_data ->
         data =
           raw_data
-          |> Enum.chunk(2)
+          |> Enum.chunk_every(2)
```
Comment (doomspork) [inline lib/deckard/build.ex:45]: `Enum.chunk/2` was deprecated.

File: lib/deckard/build.ex [hunk 2]
```diff
@@ -40,22 +71,22 @@ defmodule Deckard.Build do
...
-      :deckard
-      |> Application.get_env(__MODULE__)
-      |> Keyword.get(:url, "")
-      |> String.trim()
-      |> String.trim("/")
+  defp scan(channel, cursor \\ 0, matches \\ []) do
+    with ["0", found] <- Redis.query(["SCAN", cursor, "MATCH", "*#{channel}"]) do
+      {:ok, Enum.sort(found ++ matches, &(&1 >= &2))}
+    else
+      [cursor, found] -> scan(channel, cursor, found ++ matches)
```
Comment (doomspork) [inline lib/deckard/build.ex:79]: Redis' SCAN needs to be recursively iterated until it gives us "0" as the cursor at which point we have performed a complete scan.

File: web/views/build_view.ex [hunk 3]
```diff
@@ -1,13 +1,17 @@
 defmodule Deckard.BuildView do
-  def render("show.json", %{build: build}) do
+  def render("show.json", %{build: build, last_urgent_build: last_urgent_build}) do
     %{
-      channel: build.channel,
       build: build.build,
-      version: build.version,
+      channel: build.channel,
+      last_release_date: last_release_date(last_urgent_build),
```
Comment (btkostner) [inline web/views/build_view.ex:6]: @mmstick are you happy with this API?

**Review by btkostner — APPROVED** [review]
I approve the code and am getting ready to test it.

**Comment by doomspork** [comment]
We are waiting on these changes for now @btkostner. I need to follow up with @jackpot51 and @mmstick about our previous conversation with respects to deprecating this API possibly.

This work isn't urgent so it's okay to pause on this momentarily :ok_hand:

**Comment by doomspork** [comment]
We are moving away from this API. The urgent flag will be reflected elsewhere.

**Pull request closed without merging**